src/handlers/pug_handler.ts

```diff
@@ -74,3 +67,86 @@ export async function friends(req: any, res: express.Response) {
 
   res.render('friends', context);
 }
+
+export async function activities(req: any, res: express.Response) {
+  const log = getLogger('activities', req);
+  const context = getSharedTemplateContext(req, log);
+
+  if (!isAuthenticated(req, res, log)) {
+    return;
+  }
+
+  await syncStravaActivities(req, res, log);
+
+  context.activities = await req.context.daos.activity.findActivitiesByUser(
+    req.session.user.id,
+  );
+
+  res.render('activities', context);
+}
+async function syncStravaActivities(
+  req: any,
+  res: express.Response,
+  log: LoggerInstance,
+) {
+  const stravaProvider = new StravaProviderDAO(
+    req.session.user.id,
+    req.session.user.accessToken,
+    req.context.loggerFactory,
+  );
+
+  // get the last time activities were synced for the user. if activites have never been synced, get the activities for
+  // the past two weeks.
+  let user = await req.context.daos.user.findById(req.session.user.id);
+  const nextActivityDate = !isNullOrUndefined(user.lastActivitiesSyncedAt)
+    ? new Date(new Date(user.lastActivitiesSyncedAt).getTime())
+    : new Date(new Date().getTime() - 1209600 * 1000); // TODO: make this configurable. Currently set to past two weeks.
+
+  // get all activities from strava after the time activities were last synced.
+  const stravaActivities = await stravaProvider.getActivities(nextActivityDate);
+  _.map(stravaActivities, stravaActivity => {
+    stravaActivity.userId = req.session.user.id;
+    return stravaActivity;
+  });
+
+  let newLastSyncedAt = new Date();
+
+  // persist the activities
+  await Promise.all(
+    _.map(stravaActivities, activity => {
+      return req.context.daos.activity.create(activity);
+    }),
+  );
+
+  // set the lastActivitiesSyncedAt to now.
+  // TODO: We might want to set this to the date of the last activity, just in case a ride hasn't been uploaded to
+  // strava yet when we perform the sync.
+  await req.context.daos.user.updateLastActivitiesSyncedAt(
+    req.session.user.id,
+    newLastSyncedAt,
+  );
+}
+
+// TODO: consider moving the following functions into a base handler class
+function isAuthenticated(req: any, res: express.Response, log: LoggerInstance) {
+  if (!req.session.user) {
+    log.debug(`No user in session, redirecting`);
+    res.redirect(`/`);
+    return false;
+  }
+
+  return true;
+}
+
+function getLogger(handlerName: string, req: any) {
```

Comment:
Ooh good idea. If you put a return type on this method we'll get better typing. Eventually I'm going to fix the types around requests, but that will help stop the bleeding.